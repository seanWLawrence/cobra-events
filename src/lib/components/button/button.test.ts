import { expect, test } from 'vitest';
import { render, screen } from '@testing-library/svelte';
import '@testing-library/jest-dom';

import Button from './button.test.svelte';

test('default matches snapshot', () => {
	const name = 'some name';

	render(Button, { text: name });
	const button = screen.getByRole('button', { name });

	expect(button).toMatchSnapshot();
});

test('primary matches snapshot', () => {
	const name = 'some name';

	render(Button, { variant: 'primary', text: name });
	const button = screen.getByRole('button', { name });

	expect(button).toMatchSnapshot();
});

test('secondary matches snapshot', () => {
	const name = 'some name';

	render(Button, { variant: 'secondary', text: name });
	const button = screen.getByRole('button', { name });

	expect(button).toMatchSnapshot();
});
