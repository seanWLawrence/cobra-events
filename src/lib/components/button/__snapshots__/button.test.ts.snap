// Vitest Snapshot v1

exports[`default matches snapshot 1`] = `
<button
  class="py-4 px-6 font-mono text-2xl rounded-lg hover:shadow-none focus:shadow-none focus:outline-8 hover:outline-8 focus:outline-indigo-300 hover:outline-indigo-300 focus:outline-offset-4 hover:outline-offset-4 shadow-lg shadow-slate-300 border-2 border-indigo-600 bg-indigo-600 text-white hover:bg-indigo-700 focus:bg-indigo-700"
>
  some name
</button>
`;

exports[`primary matches snapshot 1`] = `
<button
  class="py-4 px-6 font-mono text-2xl rounded-lg hover:shadow-none focus:shadow-none focus:outline-8 hover:outline-8 focus:outline-indigo-300 hover:outline-indigo-300 focus:outline-offset-4 hover:outline-offset-4 shadow-lg shadow-slate-300 border-2 border-indigo-600 bg-indigo-600 text-white hover:bg-indigo-700 focus:bg-indigo-700"
>
  some name
</button>
`;

exports[`secondary matches snapshot 1`] = `
<button
  class="py-4 px-6 font-mono text-2xl rounded-lg hover:shadow-none focus:shadow-none focus:outline-8 hover:outline-8 focus:outline-indigo-300 hover:outline-indigo-300 focus:outline-offset-4 hover:outline-offset-4 shadow-lg shadow-slate-300 border-2 border-indigo-600 bg-transparent text-indigo-600 hover:bg-indigo-50 focus:bg-indigo-50"
>
  some name
</button>
`;
